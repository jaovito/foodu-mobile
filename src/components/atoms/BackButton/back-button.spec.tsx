import {render, fireEvent} from '@test-utils';
import {BackButton} from '.';

describe('BackButton Tests', () => {
  it('should render correctly', () => {
    const button = render(<BackButton />);

    expect(button).toBeDefined();
  });

  it('should be able to called on press', () => {
    const onPress = jest.fn();
    const button = render(<BackButton onPress={onPress} />);

    fireEvent.press(button.getByTestId('BackButtonID'));

    expect(onPress).toBeCalledTimes(1);
  });

  it('should be able to match with snapshot without onPress', () => {
    const button = render(<BackButton />).toJSON();

    expect(button).toMatchSnapshot();
  });

  it('should be able to match with snapshot with onPress', () => {
    const onPress = jest.fn();

    const button = render(<BackButton onPress={onPress} />).toJSON();

    expect(button).toMatchSnapshot();
  });
});
