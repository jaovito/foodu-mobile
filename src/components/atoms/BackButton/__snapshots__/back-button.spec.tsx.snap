// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton Tests should be able to match with snapshot with onPress 1`] = `
<RNGestureHandlerButton
  activeOpacity={0.3}
  borderless={true}
  collapsable={false}
  delayLongPress={600}
  handlerTag={4}
  handlerType="NativeViewGestureHandler"
  onActiveStateChange={[Function]}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  onPress={[MockFunction]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#FFFFFF",
      "borderRadius": 39,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 78,
      "justifyContent": "center",
      "opacity": 1,
      "width": 78,
    }
  }
  testID="BackButtonID"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 12,
        },
        [
          {
            "color": "#FFFFFF",
            "fontSize": 27,
          },
        ],
        {
          "fontFamily": "Entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</RNGestureHandlerButton>
`;

exports[`BackButton Tests should be able to match with snapshot without onPress 1`] = `
<RNGestureHandlerButton
  activeOpacity={0.3}
  borderless={true}
  collapsable={false}
  delayLongPress={600}
  handlerTag={3}
  handlerType="NativeViewGestureHandler"
  onActiveStateChange={[Function]}
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  style={
    {
      "alignItems": "center",
      "borderColor": "#FFFFFF",
      "borderRadius": 39,
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 78,
      "justifyContent": "center",
      "opacity": 1,
      "width": 78,
    }
  }
  testID="BackButtonID"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": undefined,
          "fontSize": 12,
        },
        [
          {
            "color": "#FFFFFF",
            "fontSize": 27,
          },
        ],
        {
          "fontFamily": "Entypo",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    
  </Text>
</RNGestureHandlerButton>
`;
